devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1276,27 +1292,30 @@ void HumanStateProvider::impl::selectJointsAndLinksForSecondaryCalibration(const
     }
 }
 
-void HumanStateProvider::impl::computeSecondaryCalibrationRotations(const std::vector<iDynTree::JointIndex>& jointZeroIndices, const std::vector<iDynTree::LinkIndex>& linkToCalibrateIndices)
+void HumanStateProvider::impl::computeSecondaryCalibrationRotationsForChain(const std::vector<iDynTree::JointIndex>& jointZeroIndices, const iDynTree::Transform& baseTransform, const std::vector<iDynTree::LinkIndex>& linkToCalibrateIndices)
 {
-    // setting to zero all the selected joints
+    // initialize vectors
     iDynTree::VectorDynSize jointPos(jointConfigurationSolution);
+    iDynTree::VectorDynSize jointVel(jointConfigurationSolution);
+    jointVel.zero();
```

Comment:
`jointPos` zero initialization is missing

Reply:
Initialization to zero is not required because it is already initialized to `jointConfigurationSolution` and the values is then assigned just below. For `jointVel` instead we set it to zero since we will not assign any other value later on.